ScnSoft-Education/LeetCodeTraining/LongestPalindrome/LongestPalindromeSolution.cs

```diff
@@ -0,0 +1,64 @@
+﻿namespace LeetCodeTraining.LongestPalindrome
+{
+    //https://leetcode.com/problems/longest-palindromic-substring/
+    public static class LongestPalindromeSolution
+    {
+        public static string LongestPalindrome(string s)
+        {
+            if (s.Length == 1)
+            {
+                return s;
+            }
+
+            int startIndex = 0;
+            int maxLength = 0;
+            int currLength = 0;
+            int start;
+            int end;
+
+            for (int i = 0; i < s.Length - 1; i++)
+            {
+                currLength = 1;
+                start = i - 1;
+                end = i + 1;
+
+                while (start >= 0 && end < s.Length && s[start] == s[end])
+                {
+                    currLength += 2;
+                    start--;
+                    end++;
+                }
+
+                if (currLength > maxLength)
+                {
+                    maxLength = currLength;
+                    startIndex = start + 1;
+                }
+
+                currLength = 0;
+
+                if (s[i + 1] == s[i])
+                {
+                    currLength = 2;
+                    start = i - 1;
+                    end = i + 2;
+
+                    while (start >= 0 && end < s.Length && s[start] == s[end])
```

Comment:
Please get rid of duplicated bloc of the code:
while (start >= 0 && end < s.Length && s[start] == s[end])
                {
                    currLength += 2;
                    start--;
                    end++;
                }

                if (currLength > maxLength)
                {
                    maxLength = currLength;
                    startIndex = start + 1;
                }

Reply:
By getting rid of code duplication I meant moving repeating code in a separate method/function.
But if your code changes not lead to bad performance compare to previous results, leaving current version is fine I guess. Anyway reading/understanding of the code I guess has became complicated a little.

Reply:
@shomzascnsoft thank you for feedback, since we need to check every symbol in string twice (for "\*" and possible "\*\*" central fragments of palindrome) I've decided to rewrite the code where separate cycles are responsible for each of these cases into one cycle, where i % 2 == 0 iterations are responsible for "\*" cases, and i % 2 == 1 are responsible for "\*\*" cases. I was satisfied with both memory and time complexity so I thought it's the right approach. Anyways I'll try a method approach as well, just to be sure.